client/ayon_applications/manager.py

```diff
@@ -46,6 +92,9 @@ class ApplicationManager:
             will always use these values. Gives ability to create manager
             using different settings.
     """
+    # holds connection to the process info storage
+    # - this is used to store process information about launched applications
+    _process_storage: Optional[sqlite3.Connection] = None
```

Comment:
Cna we separate the whole process storage logic from `ApplicationManager`? It should really be another class. It is bloating the manager's logic a lot making it super complicated to follow which code is related to what.